extmod/modussl_mbedtls.c

```diff
@@ -197,6 +205,14 @@ STATIC mp_uint_t socket_read(mp_obj_t o_in, void *buf, mp_uint_t size, int *errc
     mp_obj_ssl_socket_t *o = MP_OBJ_TO_PTR(o_in);
 
     int ret = mbedtls_ssl_read(&o->ssl, buf, size);
+    if (ret == MBEDTLS_ERR_SSL_PEER_CLOSE_NOTIFY) {
+        // end of stream
+        return 0;
+    }
+    if (ret == MBEDTLS_ERR_SSL_WANT_READ) {
+        *errcode = EWOULDBLOCK;
+        return 0;
+    }
```

Comment:
Hmm, no, this won't work, if you look at the full code:

The actual value of `MP_EWOULDBLOCK` is 11.

    if (ret == MBEDTLS_ERR_SSL_WANT_READ) {
        ret = MP_EWOULDBLOCK;
    }
    if (ret >= 0) {
        return ret;
    }

It can work for `socket_write`